awscli/clidriver.py

```diff
@@ -570,44 +562,50 @@ def _build_call_parameters(self, args, arg_table):
     def _unpack_arg(self, cli_argument, value):
         # Unpacks a commandline argument into a Python value by firing the
         # load-cli-arg.service-name.operation-name event.
-        session = self._service_object.session
-        service_name = self._service_object.endpoint_prefix
-        operation_name = xform_name(self._operation_object.name, '-')
+        session = self._session
+        service_name = self._operation_model.service_model.endpoint_prefix
```

Comment:
Another place  ``endpoint_prefix`` is used. I am not sure how many, if any customizations would get broken if we changed it to ``service_name``